pkg/kubelet/dockershim/helpers_windows.go

```diff
@@ -87,8 +107,14 @@ func (ds *dockerService) determinePodIPBySandboxID(sandboxID string) string {
 		// Todo: Add a kernel version check for more validation
 
 		if networkMode := os.Getenv("CONTAINER_NETWORK"); networkMode == "" {
-			// Do not return any IP, so that we would continue and get the IP of the Sandbox
-			ds.getIP(sandboxID, r)
+			if r.HostConfig.Isolation == hypervIsolationAnnotationValue {
+				if containerIP := ds.getIP(c.ID, r); containerIP != "" {
```

Comment:
Can we add a comment here saying "Return the first non-sandbox container IP as POD IP for Hyper-V container"?

Reply:
ack